### DISCUSSION TITLE
Error on `from asm3.__version__ import BUILD` at startup

### DISCUSSION THREAD

**Discussion by mcrosson** [sheltermanager/asm3#1466] [Q&A / Help] [answered]
Does anyone have an example of  `asm3.__version__.py` in order for the `main.py` import of the build string to work correctly?

I'm getting an error on run on the `main.py from asm3.__version__ import BUILD` line of code.

**Comment by bobintetley** [comment, marked as answer]
Seems very strange that you wouldn't have this? How did you do your install? There should be an ```asm3/__version__.py``` file included in both our source tarball and the debian package on our website.

If you have a working asm3 source tree, you can use "make version" to create the file.

The file just contains two string constants BUILD and VERSION, eg:

#!/usr/bin/env python3
VERSION = "48 [Wed  1 May 11:00:26 BST 2024]"
BUILD = "0501160026"

**Comment by mcrosson** [comment]
Thank you for the insights, I missed `make version` when I quickly reviewed the make file. This seems to have sorted the hiccups I ran into.

I'm using Docker for my deployment and build the container from the source tree. My process is to pull the latest code, checkout the latest tag, adjust any docker tags for the latest release and then run the below script with the below Dockerfile.

```
#!/bin/bash

cd /opt/asm3
git pull
git checkout $(git tag | sort -r | head -n1)
make version
docker build -t asm3:48 -f scripts/docker/Dockerfile .
docker tag asm3:48 asm3:latest

docker rm -f asm3

docker run -it -d \
    --name asm3 \
    --restart unless-stopped \
    -e TZ=UTC \
    -p 5000:5000
    -v /var/asm3/asm3.conf.dev:/etc/asm3.conf \
    asm3:latest
```

```
FROM ubuntu:latest

ENV DEBIAN_FRONTEND=noninteractive

ADD ./ /tmp/asm3

WORKDIR /tmp/asm3/install/deb

RUN ./makedeb.sh && rm -rf sheltermanager3

RUN apt-get update && apt-get upgrade -y \
    && apt-get install -y make python3 python3-pip python3-pil python3-webpy python3-mysqldb python3-psycopg2 \
    && apt-get install -y imagemagick \
    && apt-get install -y wkhtmltopdf \
    && apt-get install -y python3-reportlab \
    && apt-get install -y python3-requests \
    && apt-get install -y python3-boto3 \
    && apt-get install -y python3-xhtml2pdf \
    && apt-get install -y memcached \
    && apt-get install -y python3-memcache \
    && apt-get install -y libapache2-mod-wsgi-py3 \
    && apt-get install -y python3-stripe

RUN dpkg -i sheltermanager3_`cat ../../VERSION`_all.deb

WORKDIR /usr/lib/sheltermanager3
CMD python3 main.py 5000
```
